File: SistemaDecomisos/src/java/model/ActaDecomiso.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 *
 * @author Marco
 */
public class ActaDecomiso implements Serializable,Jsonable{
    private int idDecomiso;
    private Policia policia;
    private Testigo testigo;
    private Lugar lugar;
    private Date fecha;
    private String hora;
    private Interesado interesado;
    private List<Decomiso> decomisos;
    private String observaciones;

    public ActaDecomiso() {
    }

    public ActaDecomiso(int idDecomiso, Policia policia, Testigo testigo, Lugar lugar, Date fecha, String hora, Interesado interesado, List<Decomiso> decomisos, String observaciones) {
        this.idDecomiso = idDecomiso;
        this.policia = policia;
        this.testigo = testigo;
        this.lugar = lugar;
        this.fecha = fecha;
        this.hora = hora;
        this.interesado = interesado;
        this.decomisos = decomisos;
        this.observaciones = observaciones;
    }

    public int getIdDecomiso() {
        return idDecomiso;
    }

    public void setIdDecomiso(int idDecomiso) {
        this.idDecomiso = idDecomiso;
    }

    public Policia getPolicia() {
        return policia;
    }

    public void setPolicia(Policia policia) {
        this.policia = policia;
    }

    public Testigo getTestigo() {
        return testigo;
    }

    public void setTestigo(Testigo testigo) {
        this.testigo = testigo;
    }

    public Lugar getLugar() {
        return lugar;
    }

    public void setLugar(Lugar lugar) {
        this.lugar = lugar;
    }

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    public Interesado getInteresado() {
        return interesado;
    }

    public void setInteresado(Interesado interesado) {
        this.interesado = interesado;
    }

    public List<Decomiso> getDecomisos() {
        return decomisos;
    }

    public void setDecomisos(List<Decomiso> decomisos) {
        this.decomisos = decomisos;
    }

    public String getObservaciones() {
        return observaciones;
    }

    public void setObservaciones(String observaciones) {
        this.observaciones = observaciones;
    }
    
    
}

File: SistemaDecomisos/web/js/objects/ActaDecomiso.js
function ActaDecomiso(idDecomiso, policia, testigo, lugar, fecha, hora, interesado, decomisos, observaciones) {
    this.ActaDecomiso(idDecomiso, policia, testigo, lugar, fecha, hora, interesado, decomisos, observaciones);
}

ActaDecomiso.prototype = {
    idDecomiso: 0, 
    policia: "", 
    testigo: "", 
    lugar: "", 
    fecha: "", 
    hora: "", 
    interesado: "", 
    decomisos: "", 
    observaciones: "",
    ActaDecomiso: function (idDecomiso, policia, testigo, lugar, fecha, hora, interesado, decomisos, observaciones) {
        this.idDecomiso = idDecomiso, 
        this.policia = policia, 
        this.testigo = testigo, 
        this.lugar = lugar, 
        this.fecha=  fecha, 
        this.hora =  hora, 
        this.interesado = interesado, 
        this.decomisos =  decomisos, 
        this.observaciones = observaciones;
    }
};

ActaDecomiso.from = function (plain) {
    acta = new ActaDecomiso(
    plain.idDecomiso, 
    plain.policia, 
    plain.testigo, 
    plain.lugar, 
    plain.fecha, 
    plain.hora, 
    plain.interesado, 
    plain.decomisos, 
    plain.observaciones);
    return acta;
};

ActaDecomiso.to = function (acta) {
    return {
        _class: 'ActaDecomiso',
        idDecomiso: acta.idDecomiso, 
        policia: acta.policia, 
        testigo: acta.testigo, 
        lugar: acta.lugar, 
        fecha: acta.fecha, 
        hora: acta.hora, 
        interesado: acta.interesado, 
        decomisos: acta.decomisos, 
        observaciones: acta.observaciones
    };
};
 